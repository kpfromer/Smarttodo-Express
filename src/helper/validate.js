import joi from 'joi';
import boom from 'boom';
import { promisify } from 'util';
import asyncMiddleware from './async-middleware';

export default joiSchema => asyncMiddleware(async (req, res, next) => {
  return await promisify(joi.validate)(req.body, joiSchema)
    .then(validated => {
      req.validatedBody = validated;
      next();
    })
    .catch(error => 
      next(boom.badData(undefined, { validation: error.details }))
    );
})