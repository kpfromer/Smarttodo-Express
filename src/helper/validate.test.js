import joi from 'joi';
import validate from './validate';

describe('validate', () => {
  let next;
  beforeEach(() => {
    next = jest.fn();
  });
  it('handles error if data doesn\'t match joi schema', async () => {
    const schema = joi.object().keys({
      name: joi.string().required(),
      age: joi.number().positive()
    }).required();

    const req = {
      body: {
        name: '',
        age: -12451235213
      }
    }

    await validate(schema)(req, undefined, next);

    expect(next).toHaveBeenCalled();
    expect(next.mock.calls[0]).toMatchSnapshot();
  });
});