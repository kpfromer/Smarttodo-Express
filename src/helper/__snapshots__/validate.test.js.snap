// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate handles error if data doesn't match joi schema 1`] = `
Array [
  [Error: Unprocessable Entity],
]
`;
